### PR TITLE
Address most recent pr feedback

### PR DESCRIPTION
**Pull request by kldavis4**
- readme / comment updates
- make ecr repo optional
- remove datasets s3 bucket
- drop optional vpc flow logs
- make vpc / cidr blocks configurable
- output command for ingesting configuration

### CODE REVIEW

**Comment by ankur-hop** [comment]
@markhflanagan can you please take a look at this?

Kelly did us a favor and responded to some of the feedback in our PR to the Metaflow Tools repo, and was nice enough to submit it as a PR into our branch. Could you take a look at this on Monday please?

Thanks!

Ankur

File: aws/terraform/README.md [hunk 1]
```diff
@@ -57,22 +59,20 @@ Apply it:
...
-Once the terraform executes, grab the output of `metaflow_profile_json` and put it in `~/.metaflowconfig/config_prod.json`
-
-Next set the `METAFLOW_PROFILE` environment variable to `prod` to tell the metaflow client which configuration to use.
+Once the Terraform executes, configure Metaflow using `metaflow configure import ./metaflow_config_<env>_<region>.json`
 
-### Default Batch Image
+### Custom Batch Image
 
-The generated Metaflow batch configuration sets `METAFLOW_BATCH_CONTAINER_IMAGE` and `METAFLOW_BATCH_CONTAINER_REGISTRY` to an ECR repository. The Metaflow batch image must be pushed into the repository before the first flow can be executed.
+A custom batch image can be used by setting the variable `enable_custom_batch_container_registry` to `true`. This will provision an Amazon ECR registry, adn the generated Metaflow AWS Batch configuration will have `METAFLOW_BATCH_CONTAINER_IMAGE` and `METAFLOW_BATCH_CONTAINER_REGISTRY` set to the Amazon ECR repository. The Metaflow AWS Batch image must then be pushed into the repository before the first flow can be executed.
```
Comment (ankur-hop) [inline aws/terraform/README.md]: adn -> and

File: aws/terraform/README.md [hunk 2]
```diff
@@ -37,15 +37,17 @@ terraform apply --var-file prod.tfvars
...
+Copy `example.tfvars` to `prod.tfvars` (or whatever environment name you prefer) and update that `env` name and the `region` as needed. These variables are used to construct unique names for infrastructure resources.
+
+Metadata API authentication: the endpoint is exposed to the public internet via Amazon API Gateway, but only accessible to the IPs that match `access_list_cidr_blocks` (default is none).
+
 Additionally:
 * There are variables which govern the three compute environments associated with the AWS Batch queue that can be adjusted based on needs. 
-* The `enable_step_functions` flag can be set to false to not provision the step functions infrastructure. 
-* The `access_list_cidr_blocks` should be set to the network cidr blocks that will be accessing the metadata API.
+* The `enable_step_functions` flag can be set to false to not provision the AWS Step Functions infrastructure.
+* The `access_list_cidr_blocks` should be set to the network cidr blocks that will be accessing the metadata API. If you only need access from your local machine, determine your public ip address (`curl https://ifconfig.me/`) and add that value ot the list.
```
Comment (ankur-hop) [inline aws/terraform/README.md]: minor nit, but "ip" should probably be "IP"

also, "ot" -> "to"

File: aws/terraform/modules/metaflow/README.md [hunk 3]
```diff
@@ -10,9 +10,9 @@ Provides the core functionality for Metaflow which includes:
 - an API to record and query past executions (`metadata-service`)
 - orchestrated processing (`step-functions`)
 
-Depends on a VPC that has been previously set up. The output of the project `infra` is an example 
-configuration of a VPC that can be passed to this module.
+Depends on a Amazon VPC that has been previously set up. The output of the project `infra` is an example 
+configuration of a Amazon VPC that can be passed to this module.
```
Comment (ankur-hop) [inline aws/terraform/modules/metaflow/README.md]: minor, but both instances of 'a Amazon VPC' should probably be 'an Amazon VPC'

**Review by ankur-hop — COMMENTED** [review]
It's Friday afternoon so only did a super quick scan on some minor language things. Will try to look at the actual substance of the PR on Monday, unless Mark beats me to it.

File: aws/terraform/README.md [hunk 4]
```diff
@@ -37,15 +37,17 @@ terraform apply --var-file prod.tfvars
...
-The metaflow sub-project provisions the metadata API, step functions, and an AWS Batch queue. For more details see the 
+The metaflow sub-project provisions the metadata API, AWS Step Functions, and an AWS Batch queue. For more details see the 
 [README](aws/terraform/metaflow/README.md)
 
-Copy `example.tfvars` to `prod.tfvars` (or whatever environment name you prefer) and update that `env` name and the `region` as needed.
-Metadata API authentication: the endpoint is exposed to the public internet via API Gateway, but only accessible to the IPs that match `access_list_cidr_blocks` (default is none).
+Copy `example.tfvars` to `prod.tfvars` (or whatever environment name you prefer) and update that `env` name and the `region` as needed. These variables are used to construct unique names for infrastructure resources.
+
+Metadata API authentication: the endpoint is exposed to the public internet via Amazon API Gateway, but only accessible to the IPs that match `access_list_cidr_blocks` (default is none).
+
```
Comment (markhflanagan) [inline aws/terraform/README.md:46]: Does this mean the default is that the metadata service is inaccessible?

Comment (markhflanagan) [inline aws/terraform/README.md:46]: Does this mean the default is that the metadata service is inaccessible? Or is it only restricted to the whitelisted IPs if this value is provided?

File: aws/terraform/README.md [hunk 5]
```diff
@@ -57,22 +59,20 @@ Apply it:
 terraform apply --var-file prod.tfvars
 ```
 
-Once the terraform executes, grab the output of `metaflow_profile_json` and put it in `~/.metaflowconfig/config_prod.json`
-
-Next set the `METAFLOW_PROFILE` environment variable to `prod` to tell the metaflow client which configuration to use.
+Once the Terraform executes, configure Metaflow using `metaflow configure import ./metaflow_config_<env>_<region>.json`
```
Comment (markhflanagan) [inline aws/terraform/README.md:62]: Is `./metaflow_config_<env>_<region>.json` the output of terraform, like `terraform output metaflow_profile_json -raw > ./metaflow_config_<env>_<region>.json`?

If that's correct, does the following make sense?
> Once the Terraform executes, save the output of `metaflow_profile_json` to a file  with `terraform output metaflow_profile_json -raw > metaflow_config_<env>_<region>.json` and then configure Metaflow using `metaflow configure import metaflow_config_<env>_<region>.json`.

File: aws/terraform/README.md [hunk 6]
```diff
@@ -57,22 +59,20 @@ Apply it:
...
 
-### Default Batch Image
+### Custom Batch Image
```
Comment (markhflanagan) [inline aws/terraform/README.md]: Is "Custom Default AWS Batch Image" more accurate? Or "Custom AWS Batch Image Registry"?

File: aws/terraform/infra/vpc.tf [hunk 7]
```diff
@@ -11,7 +11,7 @@ resource "aws_vpc" "this" {
 }
 
 /*
- Setup a gateway between aws VPC and internet. Allow access to and from Resources
+ Setup a gateway between AWS VPC and internet. Allow access to and from resources
```
Comment (markhflanagan) [inline aws/terraform/infra/vpc.tf]: This one is "Amazon VPC", I think.

File: aws/terraform/metaflow/outputs.tf [hunk 8]
```diff
@@ -43,9 +43,9 @@ output "METAFLOW_SFN_DYNAMO_DB_TABLE" {
   description = "AWS DynamoDB table name for tracking AWS Step Functions execution metadata."
 }
 
-output "metaflow_profile_json" {
-  value       = module.metaflow.metaflow_profile_json
-  description = "Metaflow profile JSON object that can be used to communicate with this Metaflow Stack. Store this in `~/.metaflow/config_[stack-name]` and select with `$ export METAFLOW_PROFILE=[stack-name]`."
+output "metaflow_profile_configuration" {
+  value       = "metaflow configure import ${path.module}/${local.metaflow_config_filename}"
+  description = "Output the command to import Metaflow configuration"
```
Comment (markhflanagan) [inline aws/terraform/metaflow/outputs.tf]: Oh, I see. So the output is the import command with the json file path already filled in.

Does "Run this output in a shell to import the Metaflow configuration" make sense?

File: aws/terraform/modules/metaflow/modules/README.md [hunk 9]
```diff
@@ -4,7 +4,7 @@ Our Metaflow Terraform code has been separated into separate modules based on th
 
 ## Computation
 
-Sets up remote computation resources so flows can be run on EC2 instances. These resources do not perform 
+Sets up remote computation resources so flows can be run on AWS EC2 instances. These resources do not perform
```
Comment (markhflanagan) [inline aws/terraform/modules/metaflow/modules/README.md]: "AWS EC2" -> "Amazon EC2"

File: aws/terraform/modules/metaflow/variables.tf [hunk 10]
```diff
@@ -13,6 +13,12 @@ variable "enable_step_functions" {
   description = "Provisions infrastructure for step functions if enabled"
 }
 
+variable "enable_custom_batch_container_registry" {
+  type        = bool
+  default     = false
+  description = "Provisions infrastructure for custom ECR container registry if enabled"
```
Comment (markhflanagan) [inline aws/terraform/modules/metaflow/variables.tf]: "ECR" -> "Amazon ECR"?

**Review by markhflanagan — COMMENTED** [review]
Hi, thanks for adding all of these changes. Some of my comments/questions might be a little naive because I'm just jumping into how things are laid out here.

File: aws/terraform/README.md (hunk 4 above)
Comment (kldavis4) [reply aws/terraform/README.md:46]: The former. if you don't specify the access list IPs, then you won't be able to access it. I believe that the cloudformation versus provisions an authorizer, so we may want to add that in terraform as an option in a future PR.

File: aws/terraform/README.md (hunk 5 above)
Comment (kldavis4) [reply aws/terraform/README.md:62]: metaflow_config_<env>_<region>.json is a file that is created when you apply the terraform. That can then be ingested using the metaflow configure import command

File: aws/terraform/metaflow/outputs.tf (hunk 8 above)
Comment (kldavis4) [reply aws/terraform/metaflow/outputs.tf]: I updated the description and added the suggested text to the output value so it prompts the user

**Comment by kldavis4** [comment]
thanks for having a look @markhflanagan - I've made the suggested changes (and found some other instances of EC2 without Amazon)

**Comment by ankur-hop** [comment]
@markhflanagan I went ahead and merged, since it sounds like most of your comments were addressed.

**Pull request merged**